src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -155,62 +159,113 @@ subroutine SetFracIce(bounds, num_hydrologyc, filter_hydrologyc, &
+
+          qflx_top_soil           =>    waterflux_inst%qflx_top_soil_col          , & ! Output: [real(r8) (:)]  net water input into soil from top (mm/s)
+          qflx_in_soil            =>    waterflux_inst%qflx_in_soil_col           , & ! Output: [real(r8) (:)]  surface input to soil (mm/s)
+          qflx_top_soil_to_h2osfc =>   waterflux_inst%qflx_top_soil_to_h2osfc_col , & ! Output: [real(r8) (:)]  portion of qflx_top_soil going to h2osfc, minus evaporation (mm/s)
+          qflx_rain_plus_snomelt  => waterflux_inst%qflx_rain_plus_snomelt_col    , & ! Input:  [real(r8) (:)]  rain plus snow melt falling on the soil (mm/s)
+          qflx_snow_h2osfc        =>    waterflux_inst%qflx_snow_h2osfc_col       , & ! Input:  [real(r8) (:)]  snow falling on surface water (mm/s)
+          qflx_floodc             =>    waterflux_inst%qflx_floodc_col            , & ! Input:  [real(r8) (:)]  column flux of flood water from RTM
+          qflx_ev_soil            =>    waterflux_inst%qflx_ev_soil_col           , & ! Input:  [real(r8) (:)   ]  evaporation flux from soil (W/m**2) [+ to atm]
+          qflx_evap_grnd          =>    waterflux_inst%qflx_evap_grnd_col         , & ! Input:  [real(r8) (:)   ]  ground surface evaporation rate (mm H2O/s) [+]
+          qflx_ev_h2osfc          =>    waterflux_inst%qflx_ev_h2osfc_col         , & ! Input:  [real(r8) (:)   ]  evaporation flux from h2osfc (W/m**2) [+ to atm]
+
+          qflx_sat_surf           =>  surf_runoff_sat_inst%qflx_sat_surf_col      , & ! Input:  [real(r8) (:)   ]  surface runoff due to saturated surface (mm H2O /s)
+
+          frac_sno                =>    waterstate_inst%frac_sno_eff_col          , & ! Input:  [real(r8) (:)   ]  fraction of ground covered by snow (0 to 1)
+          frac_h2osfc             =>    waterstate_inst%frac_h2osfc_col          & ! Input:  [real(r8) (:)   ]  fraction of ground covered by surface water (0 to 1)
+          )
 
      do fc = 1, num_hydrologyc
         c = filter_hydrologyc(fc)
         qflx_top_soil(c) = qflx_rain_plus_snomelt(c) + qflx_snow_h2osfc(c) + qflx_floodc(c)
+
+        ! Partition surface inputs between soil and h2osfc
+        !
+        ! This is only done for soil & crop landunits, because other
+        ! hydrologically-active landunits (in particular, urban pervious road) do not use
+        ! h2osfc.
+        if (lun%itype(col%landunit(c)) == istsoil .or. lun%itype(col%landunit(c))==istcrop) then
+
+           ! explicitly use frac_sno=0 if snl=0
+           if (snl(c) >= 0) then
+              fsno=0._r8
+              ! if no snow layers, sublimation is removed from h2osoi_ice in drainage
+              qflx_evap(c)=qflx_evap_grnd(c)
+           else
+              fsno=frac_sno(c)
+              qflx_evap(c)=qflx_ev_soil(c)
+           endif
+
+           qflx_in_soil(c) = (1._r8 - frac_h2osfc(c)) * (qflx_top_soil(c)  - qflx_sat_surf(c))
```

Comment:
**Comment from @martynpclark Aug 24, 2017**

Should be `qflx_in_soil(c) = qflx_top_soil(c)  - qflx_sat_surf(c)` because forcing is constant over a grid cell.